tests/unit/cli/registry/test_cli_registry.py

```diff
@@ -0,0 +1,98 @@
+from tests.test_suite.e2e_test_system import e2e_pytest_unit
+
+
+class TestRegistry:
+    @pytest.fixture
+    def mock_templates(self, mocker):
+        template1 = mocker.MagicMock(framework="OTX001", task_type="Classification", model_template_id="001")
+        template2 = mocker.MagicMock(framework="OTX002", task_type="Segmentation", model_template_id="002")
+        template3 = mocker.MagicMock(framework="OTX001", task_type="Detection", model_template_id="003")
+        return [template1, template2, template3]
+
+    @e2e_pytest_unit
+    def test_init(self, mocker, mock_templates):
+        mock_glob = mocker.patch("glob.glob")
+        mock_glob.return_value = ["./template.yaml"]
+        mock_abspath = mocker.patch("os.path.abspath")
+        mock_abspath.return_value = "./template.yaml"
+        # mock_exists = mocker.patch("os.path.exists")
+        mock_parse_model_template = mocker.patch("otx.cli.registry.registry.parse_model_template")
+        mock_parse_model_template.return_value = mock_templates[0]
+
+        with pytest.raises(RuntimeError):
+            r = Registry()
+
+        r = Registry(templates_dir=".")
+        assert r.templates == [mock_templates[0]]
+
+        r = Registry(templates_dir=".", experimental=True)
+        assert r.templates == [mock_templates[0]] * 2
+
+        r = Registry(templates=mock_templates)
+        assert r.templates == mock_templates
+
+    @e2e_pytest_unit
+    def test_filter(self, mock_templates):
+        """Check TestRegistry.filter function working well.
+
+        <Steps>
+            1. Create an instance of Registry with the mock templates
+            2. Test filtering by framework
+            3. Test filtering by task_type
+            4. Test filtering by both framework and task_type
+        """
+        registry = Registry(templates=mock_templates)
+
+        filtered_registry = registry.filter(framework="OTX001")
+        assert len(filtered_registry.templates) == 2
+        assert filtered_registry.templates[0].task_type == "Classification"
+        assert filtered_registry.templates[1].task_type == "Detection"
+        filtered_registry = registry.filter(framework="OTX002")
+        assert len(filtered_registry.templates) == 1
+        assert filtered_registry.templates[0].task_type == "Segmentation"
+
+        filtered_registry = registry.filter(task_type="Detection")
+        assert len(filtered_registry.templates) == 1
+        assert filtered_registry.templates[0].task_type == "Detection"
+
+        filtered_registry = registry.filter(framework="OTX002", task_type="Segmentation")
+        assert len(filtered_registry.templates) == 1
+        assert filtered_registry.templates[0].task_type == "Segmentation"
+
+    @e2e_pytest_unit
+    def test_get(self, mock_templates):
+        registry = Registry(templates=mock_templates)
+        template = registry.get("003")
+        assert template.task_type == "Detection"
+
+        with pytest.raises(ValueError):
```

Comment:
I think it's better to separate this part as other test function because this is different with working well case. How do you think about?

Reply:
This test case is a test function to see if it works well. It's just checking an assert twice for the same purpose and usability. I think this could be kept as a single function, what do you think?

Reply:
Actually, I think this test includes two test case. One is getting existing one(this should works well, of course), and another one is getting non existing one. later one can be implemented variously depending on design choice (raise error, just ignore, or return others to work without error later). So, it seems better to separate them, I think. But I know this is a subjective part in some degree. If you still think they are same test case, then I'll follow your decision.

Reply:
This has been reflected.